test/functional/rpc_getmininginfo.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the getmininginfo API."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+from decimal import Decimal
+
+class GetMiningInfoTest(BitcoinTestFramework):
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 1
+        self.setup_clean_chain = False
+
+    def run_test(self):
+        mining_info = self.nodes[0].getmininginfo()
+
+        self.test_returns_all_the_keys(mining_info)
+        self.test_values(mining_info)
+        self.test_blocks_key_is_equal_to_getblockcount(mining_info)
+
+    def test_returns_all_the_keys(self, mining_info):
+        required_keys = [
+            'blocks',
+            'chain',
+            'currentblocksize',
+            'currentblocktx',
+            'currentblockweight',
+            'difficulty',
+            'errors',
+            'networkhashps',
+            'pooledtx',
+        ]
+
+        for key in required_keys:
+            assert(key in mining_info)
```

Comment:
No need to assert for keys in our tests. Python has you covered by throwing a KeyError in case you access a key that does not exist.